models/repo/setting.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"code.gitea.io/gitea/models/db"
+
+	"xorm.io/builder"
+)
+
+// Setting is a key value store of repo settings
+type Setting struct {
+	ID           int64  `xorm:"pk autoincr"`
+	RepoID       int64  `xorm:"index unique(key_repoid)"`              // to load all of someone's settings
+	SettingKey   string `xorm:"varchar(255) index unique(key_repoid)"` // ensure key is always lowercase
+	SettingValue string `xorm:"text"`
```

Comment:
Two-word column can avoid most SQL keyword conflicts.

Otherwise developers should remember to write quoted `` `key` `` again and again. There were many bugs caused by the forgotten `` ` `` before.

Reply:
Here as well:
```suggestion
	SettingKey   string `xorm:"varchar(255) index unique(key_repoid)"` // ensure key is always lowercase
	SettingValue string `xorm:"longtext"`
```